datadog-trace-utils/src/send_data/mod.rs

```diff
@@ -114,6 +115,34 @@ impl SendDataBuilder {
         self
     }
 
+    pub fn with_api_key(mut self, api_key: &str) -> SendDataBuilder {
+        self.target.api_key = Some(api_key.to_string().into());
+        self
+    }
+
+    pub fn with_retry_strategy(mut self, retry_strategy: RetryStrategy) -> SendDataBuilder {
+        self.retry_strategy = retry_strategy;
+        self
+    }
+
+    /// Returns the user defined approximate size of the data to be sent in bytes.
+    ///
+    /// # Returns
+    ///
+    /// The size of the data.
+    pub fn len(&self) -> usize {
```

Comment:
I don't think adding getters to builders is generally a good idea. The value of the builder pattern in Rust is to allow flexible setting of parameters while deferring construction until `build()` is called. This means the properties on the builder can represent an invalid state for the thing being built. It also means the values may not be set at all.

I'm not recommending it be changed as part of this PR as it already exists...but I would argue all properties should be `Option` on a builder. Size shouldn't just be `usize`. Is the size actually 0? Or have we just not set it yet on the builder?

Reply:
Thanks for the feedback! I will pass the size alongside `SendDataBuilder` in our codebase instead of calling the getter.